docs/configuration-reference/components/prometheus-operator.md

```diff
@@ -29,8 +29,10 @@ Prometheus Operator component configuration example:
 
 ```tf
 component "prometheus-operator" {
-  namespace              = "monitoring"
+  namespace = "monitoring"
+
   grafana_admin_password = "foobar"
+  grafana_ingress_host   = "grafana.mydomain.net"
```

Comment:
> Can't we use a convention such as grafana.<cluster_name>.<zone> rather than add a new knob? At the very least we could use the convention by default and allow an explicit override if this needs to be user configurable.

1. Prometheus Operator has no knowledge of the cluster it is being deployed upon. So there is no way it can enforce upon above information without having to ask user again for `cluster_name` and `zone`. This is because there is no wiring between the cluster config and the components config behind the scene. Making that work will be bigger a refactor work.

2. I am not sure if a component should be enforcing such a convention, users are totally free to select whatever they want to. But if you think it makes sense to be opionated about it then above problem still holds true.

Reply:
Yes, I'm aware we don't have this "wiring". I guess my question was "is this feasible", but I understand it's complicated and out of scope.

>I am not sure if a component should be enforcing such a convention, users are totally free to select whatever they want to. But if you think it makes sense to be opionated about it then above problem still holds true.

My point was that Prometheus/Grafana are kind of "cluster-related" and therefore it _could_ make sense to have a convention such as the one suggested above for their URLs. I even recall something about `kubectl cluster-info` showing a Grafana URL. But my main motivation for suggesting this was to reduce the number of required knobs the user needs to worry about. Adding a knob to the config is "expensive" in terms of UX.

Reply:
We can also change the whole thing to look like following:

```tf
  grafana {
    admin_password = ""

    ingress {
      host = ""

      # Needed later
      ingress_class = ""
      certmanager_cluster_issuer = ""
    }
  }
```

This is future proof as well, WDYT?

Reply:
If the user does not specify ingress block at all then ingress is not created and if user does then `host` field is mandatory.